tests/poll.rs

```diff
@@ -310,10 +311,69 @@ fn registry_operations_are_thread_safe() {
     handle3.join().unwrap();
 }
 
-// On kqueue platforms registering twice (not *re*registering) works.
+// This test checks the following reregister constraints:
+// The `reregister` arguments fully override the previous values. In other
+// words, if a socket is registered with `READABLE` interest and the call
+// to `reregister` specifies `WRITABLE`, then read interest is no longer
+// requested for the handle.
+#[test]
+fn reregister_overrides_interest() {
```

Comment:
I think the two tests can be merged; changing both the `Token` and `Interest` in a single call to reregister. If you can from `ID1`, `READABLE` to `ID2`, `WRITABLE` you can `expect_event(ID2, WRITABLE)` after using a `UdpSocket`.
Then you don't need to both with starting the listener and the barrier.

Reply:
ok, I've merged the tests, thanks